### PR TITLE
feat(CommandInteraction): add support for localized slash commands

### PR DESCRIPTION
**Pull request by suneettipirneni**
**Please describe the changes this PR makes and why it should be merged:**

Add support for localized names and descriptions in slash command payloads.

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Comment by ImRodry** [comment]
Discord API docs PR https://github.com/discord/discord-api-docs/pull/4653

**Review by ImRodry — CHANGES_REQUESTED** [review]
Missed the edit methods in ApplicationCommand (also please provide some examples in those)

File: packages/discord.js/src/managers/ApplicationCommandManager.js [hunk 1]
```diff
@@ -216,7 +216,9 @@ class ApplicationCommandManager extends CachedManager {
   static transformCommand(command) {
     return {
       name: command.name,
+      name_localizations: command.nameLocalizations,
       description: command.description,
+      description_localizations: command.descriptionLocalizations,
```
Comment (Smidul) [inline packages/discord.js/src/managers/ApplicationCommandManager.js:221]: Shouldn't both also check their variants in `snake_case`?

File: packages/discord.js/src/structures/ApplicationCommand.js [hunk 2]
```diff
@@ -190,6 +210,22 @@ class ApplicationCommand extends Base {
...
   }
 
+  /**
+   * Edits the localized names of this ApplicationCommand
+   * @param {Object<string, string>} nameLocalizations The new localized names for the command
+   * @returns {Promise<ApplicationCommand>}
+   * @example
+   * // Edit the name localizations of this command
+   * command.setLocalizedNames({
+   *   'en-US': 'ping',
```
Comment (ImRodry) [inline packages/discord.js/src/structures/ApplicationCommand.js:220]: wouldn't en-US be the one in `name`? This should probably have a different locale

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]
LGTM. However, docs state that behaviour of `GET` endpoints is changed, therefore they won't return `name_localizations` & `description_localizations` by default, instead they will return `name_localized` & `description_localized`, which are not included in the PR. The way this behaviour could be altered is by supplying `?with_localizations=true`. Not sure what's the best solution here though.

**Comment by suneettipirneni** [comment]
> LGTM. However, docs state that behaviour of `GET` endpoints is changed, therefore they won't return `name_localizations` & `description_localizations` by default, instead they will return `name_localized` & `description_localized`, which are not included in the PR. The way this behaviour could be altered is by supplying `?with_localizations=true`. Not sure what's the best solution here though.

I can add it as an option in the `ApplicationCommandManager`

**Comment by FedorLap2006** [comment]
> > LGTM. However, docs state that behaviour of `GET` endpoints is changed, therefore they won't return `name_localizations` & `description_localizations` by default, instead they will return `name_localized` & `description_localized`, which are not included in the PR. The way this behaviour could be altered is by supplying `?with_localizations=true`. Not sure what's the best solution here though.
> 
> I can add it as an option in the `ApplicationCommandManager`

Problem is, when it's set to false (i.e. endpoint provides `localized` fields) you would need to pass an additional header to alter the language. Which I'm not sure where would be the best to put

File: packages/discord.js/src/structures/ApplicationCommand.js [hunk 3]
```diff
@@ -70,6 +80,16 @@ class ApplicationCommand extends Base {
       this.description = data.description;
     }
 
+    if ('description_localizations' in data) {
+      /**
+       * The description localizations for this command
+       * @type {?Object<string, string>}
```
Comment (xHyroM) [inline packages/discord.js/src/structures/ApplicationCommand.js:86]: You can use `LocaleString` from `discord-api-types`

**Review by xHyroM — CHANGES_REQUESTED** [review]

**Review by mafineeek — APPROVED** [review]

File: packages/discord.js/src/managers/ApplicationCommandManager.js [hunk 4]
```diff
@@ -92,19 +94,26 @@ class ApplicationCommandManager extends CachedManager {
    *   .then(commands => console.log(`Fetched ${commands.size} commands`))
    *   .catch(console.error);
    */
-  async fetch(id, { guildId, cache = true, force = false } = {}) {
+  async fetch(id, { guildId, cache = true, force = false, locale, withLocalizations } = {}) {
+    const fetchOptions = {
+      headers: {
+        'X-Discord-Locale': locale,
+      },
+      query: withLocalizations ? new URLSearchParams({ with_localizations: withLocalizations }) : undefined,
```
Comment (vladfrangu) [inline packages/discord.js/src/managers/ApplicationCommandManager.js:102]: Nit
```suggestion
      query: typeof withLocalizations === 'boolean' ? new URLSearchParams({ with_localizations: withLocalizations }) : undefined,
```

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Comment by kyranet** [comment]
There are merge conflicts, please update or rebase your branch.

File: packages/discord.js/src/managers/ApplicationCommandManager.js [hunk 5]
```diff
@@ -74,6 +74,8 @@ class ApplicationCommandManager extends CachedManager {
    * Options used to fetch Application Commands from Discord
    * @typedef {BaseFetchOptions} FetchApplicationCommandOptions
    * @property {Snowflake} [guildId] The guild's id to fetch commands for, for when the guild is not cached
+   * @property {LocaleString} [locale] The locale to use when fetching this command
+   * @property {boolean} [withLocalizations] Whether or not to fetch all localization data
```
Comment (Jiralite) [inline packages/discord.js/src/managers/ApplicationCommandManager.js:78]:
```suggestion
   * @property {boolean} [withLocalizations] Whether to fetch all localization data
```

**Pull request merged**